Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -342,6 +342,16 @@ String WebAutomationSession::handleForWebFrameID(std::optional<FrameIdentifier>
     return handle;
 }
 
+String WebAutomationSession::effectiveHandleForWebFrameProxy(const WebFrameProxy& webFrameProxy)
+{
+    if (webFrameProxy.isMainFrame()) {
+        if (RefPtr page = webFrameProxy.page())
+            return handleForWebPageProxy(*page);
+    }
+
+    return handleForWebFrameID(webFrameProxy.frameID());
+}
+
```

Comment:
This seems redundant with the below, why not combine them? Are there some cases where we rely on empty string for main frame? (Seems unlikely)